src/components/FloatingActionButton/FabPlusIcon.js

```diff
@@ -0,0 +1,49 @@
+import PropTypes from 'prop-types';
+import React, {useEffect} from 'react';
+import Animated, {Easing, interpolateColor, useAnimatedProps, useSharedValue, withTiming} from 'react-native-reanimated';
+import Svg, {Path} from 'react-native-svg';
+import useTheme from '@hooks/useTheme';
+
+const AnimatedPath = Animated.createAnimatedComponent(Path);
+
+const propTypes = {
+    /* Current state (active or not active) of the component */
+    isActive: PropTypes.bool.isRequired,
+};
+
+function FabPlusIcon({isActive}) {
+    const theme = useTheme();
+    const animatedValue = useSharedValue(isActive ? 1 : 0);
+
+    useEffect(() => {
+        animatedValue.value = withTiming(isActive ? 1 : 0, {
+            duration: 340,
+            easing: Easing.inOut(Easing.ease),
+        });
+    }, [isActive, animatedValue]);
+
+    const animatedProps = useAnimatedProps(() => {
+        const fill = interpolateColor(animatedValue.value, [0, 1], [theme.textLight, theme.textDark]);
```

Comment:
FYI if you use `foo.bar.baz` inside a worklet, the whole object `foo` is captured and copied to the UI runtime. In some cases it's good (maybe even here, since we cache copied objects) but usually it's safer to pass the leaf nodes, e.g.:
```ts
// this is regular React code
const { baz } = foo.bar;

useAnimatedProps(() => {
  // this is a worklet
  // use `baz` directly
});
```